### PR TITLE
Fixed #27023 -- ~/.pythonrc.py loaded twice if PYTHONSTARTUP set to ~/.pythonrc.py

### PR DESCRIPTION
**Pull request by inglesp**

### CODE REVIEW

File: django/core/management/commands/shell.py [hunk 1]
```diff
@@ -88,10 +88,15 @@ def python(self, options):
         # We want to honor both $PYTHONSTARTUP and .pythonrc.py, so follow system
         # conventions and get $PYTHONSTARTUP first then .pythonrc.py.
         if not options['no_startup']:
-            for pythonrc in (os.environ.get("PYTHONSTARTUP"), '~/.pythonrc.py'):
-                if not pythonrc:
-                    continue
-                pythonrc = os.path.expanduser(pythonrc)
+            if os.environ.get('PYTHONSTARTUP'):
```
Comment (timgraham) [inline django/core/management/commands/shell.py]: A bit simpler?
```python
python_startup = os.environ.get('PYTHONSTARTUP')
python_rc = os.path.expanduser('~/.pythonrc.py')
python_rcs = [os.path.expanduser(python_startup)] if python_startup else []
if python_rc != python_startup:
    python_rcs.append(python_rc)
```

**Comment by inglesp** [comment]
Thanks for the feedback @timgraham.

It turns out that any environment variable will already be shell-expanded, so I've just made an even simpler fix.

**Comment by timgraham** [comment]
I also thought of using a set but this won't guarantee ordering which based on the comment, seems to be important.

**Comment by inglesp** [comment]
Thanks @timgraham -- I've fixed this with an OrderedSet.

Looking into the behaviour of the Python interpreter at startup, it seems that `~/.pythonrc.py` is a hangover from the `site` module which was [deprecated in 2.6](https://docs.python.org/2/library/user.html).

As such, it could be argued that Django should drop the special-casing of `~/.pythonrc.py`, and just use whatever `$PYTHONSTARTUP` refers to, but that's a separate discussion.

**Comment by timgraham** [comment]
merged in e139ef5741716d8cb715ff17da2b33df08d85309, thanks!

**Pull request closed without merging**